src/app/modules/ix-forms/components/ix-dynamic-form/ix-dynamic-form-item/ix-dynamic-form-item.component.spec.ts

```diff
@@ -1,89 +1,254 @@
+      spectator.component.dynamicSchema = ipaddrSchema;
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-ip-input-with-netmask')).toBeVisible();
+      expect(spectator.query('ix-ip-input-with-netmask')).not.toBeDisabled();
+
+      expect(spectator.query('ix-ip-input-with-netmask')).not.toBeHidden();
+      spectator.component.dynamicForm.controls.ipaddr.disable();
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-ip-input-with-netmask')).toBeHidden();
+    });
+
+    it('explorer', () => {
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = explorerSchema;
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-explorer')).toBeVisible();
+      expect(spectator.query('ix-explorer')).not.toBeDisabled();
+
+      expect(spectator.query('ix-explorer')).not.toBeHidden();
+      spectator.component.dynamicForm.controls.explorer.disable();
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-explorer')).toBeHidden();
+    });
+
+    it('list', () => {
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = listSchema;
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-list')).toBeVisible();
+      expect(spectator.query('ix-list')).not.toBeDisabled();
+
+      expect(spectator.query('ix-list')).not.toBeHidden();
+      spectator.component.dynamicForm.controls.list.disable();
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-list')).toBeHidden();
+    });
+
+    it('dict', () => {
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = dictSchema;
+      spectator.detectComponentChanges();
+      expect(spectator.query('.label')).toHaveText('Label Dict');
+
+      expect(spectator.queryAll('ix-dynamic-form-item').length).toEqual(3);
+
+      spectator.queryAll('ix-dynamic-form-item').forEach((item) => {
+        expect(item).not.toBeDisabled();
+        expect(item).not.toBeHidden();
+      });
+
+      spectator.component.dynamicForm.controls.dict.disable();
+      spectator.detectComponentChanges();
+
+      spectator.queryAll('ix-dynamic-form-item').forEach((item) => {
+        expect(item).not.toBeDisabled();
+        expect(item).toBeHidden();
+      });
+    });
   });
 
-  describe('Control Emit', () => {
+  describe('Component methods', () => {
+    it('isHidden()', () => {
+      dynamicForm.controls.list.disable();
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = listSchema;
+      expect(spectator.component.isHidden).toBeTruthy();
+    });
+
+    it('getFormArray()', () => {
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = listSchema;
+      expect(spectator.component.getFormArray).toEqual(dynamicForm.controls.list);
+    });
```

Comment:
I don't see value in testing this. `getFormArray` is public only because Angular doesn't allow private methods to be called in templates.
In spirit this method is private and is not meant to be interacted with by others, i.e. it's not really part of the public interface of this component.
How you do interact with this method is through a template. Ix-list renders a list of items and this is what needs to be checked in 'ix-list' case above.